extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityUtils.java

```diff
@@ -18,4 +23,26 @@ public static AuthenticationRequest setRoutingContextAttribute(AuthenticationReq
     public static RoutingContext getRoutingContextAttribute(AuthenticationRequest request) {
         return request.getAttribute(ROUTING_CONTEXT_ATTRIBUTE);
     }
+
+    /**
+     * Add {@link RoutingContext} to Vert.x duplicated context local data.
+     */
+    public static void setRoutingContextAttribute(RoutingContext event) {
+        final Context context = Vertx.currentContext();
+        if (isSafeToUseContext(context)) {
```

Comment:
it's just ignored? we don't want to throw an exception here?

Reply:
That's a good question. Situation is that I expect this always to be true at this moment because in `VertxHttpRecorder` the duplicated context is enforced, however I won't risk an exception and I think it is safer to always check. 

So no, I acknowledge there can be extensions that for some reason will try to use root context and then feature that relies on duplicated context won't work. Please note that documentation explicitly mention this feature depends on duplicated context so users can check https://quarkus.io/guides/duplicated-context#context-local-data and see if they can use it or not.

Reply:
If `setRoutingContextAttribute` is exposed to external users, to me, would be better to check it and fail/report somehow what's going on.
if the users are "internals" - still other known extensions we have control of and test coverage, you can just use an `assert` which should be enabled on tests IIRC (@geoand can confirm, I'm not sure...and TBH not everyone love asserts, so it's a matter of personal preferences...)

Reply:
But test coverage won't help here, we can't control if some user code or maybe Quarkiverse extension will mark https://github.com/quarkusio/quarkus/blob/447067819a58a10841375f41c12c36fb0eef5dbf/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/context/VertxContextSafetyToggle.java#L101 as unsafe or will enforce root context.

This feature makes sense because duplicated context should be used in most cases and only some rare cases will not have `RoutingContext` available.

Is this maybe problem of documentation, do you think I didn't highlight it enough that this feature depends on duplicated context? Please let me know.

Reply:
> If setRoutingContextAttribute is exposed to external users, to me, would be better to check it and fail/report somehow what's going on.

Okay, I've added extra logging for this situation, but please not this is best effort feature which as far as my knowledge of duplicated context goes, will be available for most users. Does it look better @franz1981 ?